website/notifications/emails.py

```diff
@@ -52,25 +52,40 @@ def notify(event, user, node, timestamp, **context):
     return sent_users
 
 def notify_mentions(event, user, node, timestamp, **context):
+    recipient_ids = context.get('new_mentions', [])
+    recipients = OSFUser.objects.filter(guids___id__in=recipient_ids)
+    sent_users = notify_global_event(event, user, node, timestamp, recipients, context=context)
+    return sent_users
+
+def notify_global_event(event, sender_user, node, timestamp, recipients, template=None, context=None):
     event_type = utils.find_subscription_type(event)
     sent_users = []
-    new_mentions = context.get('new_mentions', [])
-    for m in new_mentions:
-        mentioned_user = OSFUser.load(m)
-        subscriptions = get_user_subscriptions(mentioned_user, event_type)
+
+    # Initialize the subscriptions dict
+    users_subscriptions = {}
+    for key in constants.NOTIFICATION_TYPES:
+        users_subscriptions[key] = []
+
+    # Group recipients IDs per each notification type
+    # e.g. {'email_transactional': [u'vsu7t', u'evz43'], 'none': [], 'email_digest': []}
+    for recipient in recipients:
+        subscriptions = get_user_subscriptions(recipient, event_type)
         for notification_type in subscriptions:
-            if (
-                notification_type != 'none' and
-                subscriptions[notification_type] and
-                m in subscriptions[notification_type]
-            ):
-                store_emails([m], notification_type, 'mentions', user, node,
-                                 timestamp, **context)
-                sent_users.extend([m])
+            if (notification_type != 'none' and subscriptions[notification_type] and recipient._id in subscriptions[notification_type]):
```

Comment:
If you do use `set`s, it might be cleaner to get all the guids together first:
```py
for notification_type, user_guids in subscriptions.items():
    users_subscriptions[notification_type].update(user_guids)
```
then limit to the recipients below:
```py
for notification_type, user_ids in users_subscriptions.items():
    send_to = user_ids.intersection(recipient_ids)
    sent_users.extend(send_to)
    store_emails(...)
```
or something like that. Just a matter of taste, but I want to encourage you to rethink the logic that was already there, and always try to leave code cleaner than you found it.